agent/connect/ca/provider_vault.go

```diff
@@ -864,6 +864,36 @@ func (v *VaultProvider) setNamespace(namespace string) func() {
 	}
 }
 
+// autotidyIssuers sets Vault's auto-tidy to remove expired issuers
+// Returns a boolean on success for testing (as there is no post-facto way of
+// checking if it is set). Logs at info level on failure to set and why, returning the
+// log message for test purposes as well.
+func (v *VaultProvider) autotidyIssuers(path string) (bool, string) {
+	s, err := v.client.Logical().Write(path+"/config/auto-tidy",
+		map[string]interface{}{
+			"enabled":              true,
+			"tidy_expired_issuers": true,
+		})
```

Comment:
Do you know if this clobbers any existing configuration from either `config/tidy` or `config/auto-tidy` endpoints?

When I implemented `v.setDefaultIntermediateIssuer`, I was told that I should do a "read-then-write" because the Vault API doesn't have PUT/PATCH semantics.

We should do a check that any extra config that's already set in Vault is respected.

Reply:
No, it doesn't clobber previous settings. `config/auto-tidy` works as an update, only changing those values that it explicitly assigns. And `config/tidy` doesn't persist settings, it is called to trigger a tidy based on the fields sent that time. Also note that there is no GET for the auto-tidy settings, you only get them back when you set them.